sdk/nodejs/api/utils.ts

```diff
@@ -8,17 +8,25 @@ import {
   NotAwaitedRequestError,
   ExecError,
 } from "../common/errors/index.js"
-import { QueryTree } from "./client.gen.js"
+import { isEnum, QueryTree } from "./client.gen.js"
 
 /**
  * Format argument into GraphQL query format.
  */
 function buildArgs(args: any): string {
   // Remove unwanted quotes
-  const formatValue = (value: string) =>
-    JSON.stringify(value).replace(/\{"[a-zA-Z]+":|,"[a-zA-Z]+":/gi, (str) =>
-      str.replace(/"/g, "")
+  const formatValue = (value: string) => {
+    if (isEnum(value as never)) {
```

Comment:
What would happen if I tried to pass the string value `"TCP"` to something? Seems like it would get converted into `TCP` because it happens to match one of our enums.

Reply:
That's a good question, I think we cannot handle that for now :/ we might need to use an actual query builder as you mention there: https://github.com/dagger/dagger/pull/5594#pullrequestreview-1568038105

Reply:
IMO this would normally be a blocker because it will almost certainly leave a user blocked someday. Basically these values become "banned words":

* "TCP"
* "UDP"
* "SHARED"
* "LOCKED"
* "PRIVATE"
* "EStarGZ"
* "Gzip"
* "Uncompressed"
* "Zstd"
* "DockerMediaTypes"
* "OCIMediaTypes"

It seems OK for now to put the fire out since it's the lesser evil to enums being completely broken, but I think we should prioritize fixing it properly ASAP.